### PR TITLE
Publish build scan for commit distribution build

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test PT

File: build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt [hunk 1]
```diff
@@ -80,6 +80,7 @@ abstract class BuildCommitDistribution @Inject internal constructor(
         }
         execOps.exec {
             commandLine(*getBuildCommands())
+            environment(getBuildEnvs())
```
Comment (wolfs) [inline build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt:83]: What did we pass here before? `System.getenv())`? Are we losing any environment variables now?

File: build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt [hunk 2]
```diff
@@ -96,18 +97,29 @@ abstract class BuildCommitDistribution @Inject internal constructor(
...
+
+        if (project.gradle.startParameter.isBuildCacheEnabled) {
             val buildCacheConf = (project.gradle as GradleInternal).settings.buildCache
             val remoteCache = buildCacheConf.remote as HttpBuildCache?
             if (remoteCache?.url != null) {
-                buildCommands.add("-Dgradle.cache.remote.url=${remoteCache.url}")
-                buildCommands.add("-Dgradle.cache.remote.username=${remoteCache.credentials.username}")
+                envs["GRADLE_CACHE_REMOTE_URL"] = remoteCache.url.toString()
+                envs["GRADLE_CACHE_REMOTE_USERNAME"] = remoteCache.credentials.username!!
+                envs["GRADLE_CACHE_REMOTE_PASSWORD"] = remoteCache.credentials.password!!
```
Comment (wolfs) [inline build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt:117]: IIUC, the password is for pushing? Do we need to pass the `CI` environment variable as well so pushing is on? Or does this happen automatically?

**Review by wolfs — APPROVED** [review]
I think this looks good, I left some questions.

File: build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt [hunk 3]
```diff
@@ -96,18 +97,29 @@ abstract class BuildCommitDistribution @Inject internal constructor(
             "-PtoolingApiShadedJarInstallPath=" + commitDistributionToolingApiJar.get().asFile.absolutePath,
```
Comment (wolfs) [inline build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt:97]: I think we should tag these builds. We can do this by passing a system property.

**Comment by wolfs** [comment]
Ok, looks like I don't understand this change. I found https://ge.gradle.org/s/3i3vb66juaiee, which is a build scan for the build commit distribution build. So the scan is already published. What is this change about? Pushing to the remote build cache as well?

**Comment by blindpirate** [comment]
OK, I misread something. I thought the environment variable propagation limitation was applied for all forked processes, but actually it's only for `Test` tasks: https://github.com/gradle/gradle/blob/7fcdf31df1867f39e0836049592ef9872208d4bb/build-logic/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt#L155 

We're good enough already.

**Pull request closed without merging**